### PR TITLE
Release warning test

### LINKED ISSUE #2632: CircleCI test fails to capture compilation errors/warnings in "release" mode
The CircleCI configuration looks like the following, which configures the project in "debug" mode, which means it will fail to capture any error/warning arising in "release" mode. 
```
#!/bin/bash -eo pipefail
cd build
cmake -DBUILD_TEST_APPS=ON ..
make -j7 &> /tmp/output.txt
```

### PR DESCRIPTION
**Pull request by sgpearse**
This PR adds a test for warnings on Ubuntu's release build.  Fixes #2632.

### CODE REVIEW

**Comment by shaomeng** [comment]
@sgpearse will implement a Release Mode warning/error test on centos7 as well.

File: .circleci/config.yml [hunk 1]
```diff
@@ -29,10 +29,15 @@ jobs:
           command: |
             cd build
             cmake -DBUILD_TEST_APPS=ON ..
-            make -j7 &> /tmp/output.txt
+            make -j7 &> /tmp/debugOutput.txt
+            cmake -DBUILD_TEST_APPS=ON -DCMAKE_BUILD_TYPE=Release ..
```
Comment (shaomeng) [inline .circleci/config.yml:33]: I suggest you add a `make clean` before a new configuration and a new build.

**Review by shaomeng — COMMENTED** [review]
Question to Scott: Does this script test errors that might arise?

File: .circleci/config.yml (hunk 1 above)
Comment (shaomeng) [reply .circleci/config.yml:34]: Address in commit 95ceb57

**Comment by sgpearse** [comment]
> Question to Scott: Does this script test errors that might arise?

@shaomeng Yes, if the build fails we get a big red X.  Here's an example:

https://github.com/NCAR/VAPOR/compare/releaseWarningTest_errorTest?expand=1

**Review by StasJ — APPROVED** [review]

**Comment by shaomeng** [comment]
> > Question to Scott: Does this script test errors that might arise?
> 
> @shaomeng Yes, if the build fails we get a big red X. Here's an example:
> 
> https://github.com/NCAR/VAPOR/compare/releaseWarningTest_errorTest?expand=1

Great to know, thanks Scott!

**Comment by sgpearse** [comment]
@shaomeng Centos is now being tested for warnings in release and debug mode.

**Review by StasJ — APPROVED** [review]

**Review by shaomeng — APPROVED** [review]

**Pull request merged**